server/models/Request.js

```diff
@@ -0,0 +1,24 @@
+import mongoose from "mongoose";
+
+const { Schema, model } = mongoose;
+
+const requestSchema = new mongoose.Schema(
+  {
+    customer: { type: Schema.Types.ObjectId, ref: "User", required: true },
+    provider: { type: Schema.Types.ObjectId, ref: "User", required: true },
+    start: { type: Date, required: true },
+    end: { type: Date, required: true },
```

Comment:
As a bonus, you could look into adding validation here to make sure `end > start` 😎